File: 0920-uncommon-words-from-two-sentences/0920-uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        ans = []
        str1 = s1.split(" ")
        str2 = s2.split(" ")
        


        for i in str1:
            if str1.count(i) == 1:
                if i not in str2:
                    ans.append(i)
        for i in str2:
            if str2.count(i) == 1:
                if i not in str1:
                    ans.append(i)
                    
        return ans

File: 0920-uncommon-words-from-two-sentences/0920-uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
      String[] str1=s1.split(" ");
      String[] str2=s2.split(" ");

      HashMap<String,Integer> hm=new HashMap<>();

      for(String s:str1){
        hm.put(s,hm.getOrDefault(s,0)+1);
      }
      for(String s:str2){
        hm.put(s,hm.getOrDefault(s,0)+1);
      }

      int count=0;
      for(Map.Entry<String,Integer> e:hm.entrySet()){
        if(e.getValue()==1){
            count++;
        }
      }
      String[] arr= new String[count];
      int k=0;

      for(Map.Entry<String,Integer>e:hm.entrySet()){
        if(e.getValue()==1){
            arr[k++]=e.getKey();
        }
      }
      return arr;
    }
}